src/ducks/Alert/components/AlertStepsSelector/AlertStepsSelector.js

```diff
@@ -0,0 +1,58 @@
+import React, { useCallback, useEffect } from 'react'
+import Steps from '../../../../components/Steps/Steps'
+
+function AlertStepsSelector ({
+  size,
+  selectedType,
+  selectedStep,
+  setSelectedStep,
+  isMetricsDisabled,
+  items,
+  visitedSteps,
+  setVisitedSteps
+}) {
+  useEffect(() => {
+    if (selectedStep === undefined) {
+      setVisitedSteps([])
+    }
+  }, [selectedType])
+
+  const handleStepClick = stepIndex => {
+    setSelectedStep(stepIndex)
+
+    if (!visitedSteps.includes(stepIndex)) {
+      setVisitedSteps(prev => [...prev, stepIndex])
+    }
+  }
+
+  const renderSteps = useCallback(() => {
+    return items.map((step, index) => {
+      const disabled = items.length !== 3 && isMetricsDisabled && index === 1
+
+      let status = (visitedSteps.includes(index) && 'finish') || 'process'
+
+      if (items.length !== 3 && isMetricsDisabled && index === 0) {
+        status = 'process'
+      }
```

Comment:
1. `items.length !== 3 && isMetricsDisabled` can be reusing by storing it in a variable;
2. Convert `visitedSteps` to the `Set` (using useMemo) for `O(1)` lookup. Can be reused on `LOC 23`